postgres_stats.c

```diff
@@ -242,27 +242,39 @@ static void merge_stats(pg_stat_list *pg_stats, proc_stat_list proc_stats)
 
 static void read_proc_cmdline(pg_stat *stat)
 {
-	char buf[256];
+	char buf[MAXPGPATH];
 	FILE *f = open_proc_file(stat->pid, "cmdline");
 	if (f == NULL) return;
 
 	if (fgets(buf, sizeof(buf), f) && strncmp(buf, "postgres: ", 10) == 0) {
-		const char *cmd = buf + 10;
-		char *p = strstr(cmd, " process");
-		if (stat->ps.cmdline == NULL) {
-			if (p == NULL)
-				stat->ps.cmdline = pstrdup("unknown");
-			else {
-				*p = '\0';
-				stat->ps.cmdline = pstrdup(cmd);
-			}
-		}
+		char *p, *lp = NULL;
+
+		// cluster name can contain ' process ' string, so we need to skip all but last
```

Comment:
I would think nobody is realistically going use a cluster name with substring `" process "` in it.